app/js/services/DiseaseService.js

```diff
@@ -26,67 +42,68 @@ function DiseaseService($q, $resource, AppSettings, DiseaseModel, $log, filterFi
    *
    * @return {Array} filtered array of DiseaseModels
    */
-  service.query = (searchQuery, mutationsGenes)=>{
-    
-    
-      $log.log(`query:${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`);
-
-
-      let diseasePromise = new Promise((resolve, reject)=>{
-        if ($localStorage.diseaseData && $localStorage.diseaseData.count) {
-          if($localStorage.diseaseData.count === 0) return  reject(`No Disease Types found matching: "${searchQuery}"`);
-          $log.log('fetch disease data from localStorage',$localStorage.diseaseData);
-           resolve($localStorage.diseaseData); 
-        }
-
-        $log.log('fetch disease data from DB resource');
-
-        DISEASES_RESOURCE.query((diseaseResponse) => {
+  service.query = (searchQuery, mutationsGenes) => {
+    $log.log(`query:${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`);
+
+    let diseasePromise = new Promise((resolve, reject) => {
+      if ($localStorage.diseaseData && $localStorage.diseaseData.count) {
+        if ($localStorage.diseaseData.count === 0)
+          return reject(`No Disease Types found matching: "${searchQuery}"`);
+        $log.log(
+          "fetch disease data from localStorage",
+          $localStorage.diseaseData
+        );
+        resolve($localStorage.diseaseData);
+      }
+
+      $log.log("fetch disease data from DB resource");
+
+      DISEASES_RESOURCE.query(
+        diseaseResponse => {
           $log.log(`found ${diseaseResponse.count} disease types in DB`);
-          if(diseaseResponse.count === 0) return  reject(`No Disease Types found matching: "${searchQuery}"`);
+          if (diseaseResponse.count === 0)
+            return reject(`No Disease Types found matching: "${searchQuery}"`);
 
           $localStorage.diseaseData = _.assign({}, diseaseResponse);
           resolve(diseaseResponse);
-        }, error => {
-          console.log(error)
+        },
+        error => {
+          console.log(error);
           NotificationService.notify({
-            type:    'error',
+            type: "error",
             message: `Failed to load diseases.`
           });
```

Comment:
If this error condition arises, I think you also need to have a `return reject();` here similar to line 65 above or the promise will hang.